### PR TITLE
feat(provider/kubernetes): v2 Add find artifacts from resource stage

### PR DESCRIPTION
**Pull request by wjoel**
See https://github.com/spinnaker/orca/pull/1914

We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

**Comment by lwander** [comment]
https://github.com/spinnaker/spinnaker/issues/2267

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceStage.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { KUBERNETES_MANIFEST_SELECTOR } from '../../../manifest/selector/selector.component';
+
+export const KUBERNETES_FIND_ARTIFACTS_FROM_RESOURCE_STAGE = 'spinnaker.kubernetes.v2.pipeline.stage.findArtifactsFromResource';
+
+module(KUBERNETES_FIND_ARTIFACTS_FROM_RESOURCE_STAGE, [
+  PIPELINE_CONFIG_PROVIDER,
+  KUBERNETES_MANIFEST_SELECTOR
+]).config((pipelineConfigProvider: PipelineConfigProvider) => {
+    // Todo: replace feature flag with proper versioned provider mechanism once available.
+    if (SETTINGS.feature.versionedProviders) {
```
Comment (lwander) [inline app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceStage.ts]: I would feature flag this on the "artifacts" feature flag instead.

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceConfig.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  <expected-artifact-multi-selector command="ctrl.$scope.stage"
+                                    ids-field="expectedArtifactIds"
+                                    artifact-label="Artifacts To Bind"
+                                    expected-artifacts="ctrl.expectedArtifacts">
+  </expected-artifact-multi-selector>
+
+  <div class="container-fluid form-horizontal">
+    <div class="form-group">
+      <div class="col-md-3 sm-label-right">
+        Keep default artifacts
```
Comment (lwander) [inline app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceConfig.html]: Can you had a help label describing what this does? Here are some examples: https://github.com/spinnaker/deck/pull/4723/files

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceStage.ts (hunk 1 above)
Comment (wjoel) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceStage.ts]: Done.

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceConfig.html (hunk 2 above)
Comment (wjoel) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/findArtifactsFromResource/findArtifactsFromResourceConfig.html]: Removed in the simplified version (perhaps to return in a modified form in the future).

**Pull request merged**